app/src/main/java/com/example/atheneum/activities/AddEditBookActivity.java

```diff
@@ -217,6 +250,41 @@ public boolean allFieldsFilled() {
         return allFilled;
     }
 
+    public void updateBook(){
+        Log.i(TAG, "UpdateBook*** Save Button Pressed");
+
+        if (!allFieldsFilled()) {
+            // TODO : Display Error Prompt
+            Log.w(TAG, "AddBook*** Error fields unfilled");
+            Toast.makeText(context, "Cannot save with unfilled fields!", Toast.LENGTH_SHORT).show();
+
+            // call all input validations to show error prompts
+            for (EditTextWithValidator editTextVal : editTextWithValidatorArray) {
+                editTextVal.validator.validate(editTextVal.editText);
+            }
+            return;
+        }
+        Log.i(TAG, "UpdateBook*** Fields checked, all filled");
+
+        if (FirebaseAuthUtils.isCurrentUserAuthenticated()) {
```

Comment:
Why are you creating another instance of the BookInfoViewModel here? Ideally you only need one instance of the BookInfoViewModel per activity since ViewModelProviders.of() will only give you one instance of a BookInfoViewModel per userID you pass into it's factory class. I think it would be better to just have the instance of BookInfoViewModel to be private and part of the activity and so you can share it in other places.

Reply:
Also, BookInfoViewModel takes in a bookID so why did you pass in an ownerID to it?

Reply:
Shoot sorry, totally missed the other instance of BookInfoViewModel. Changed it now.